tests/virtual_machines_test.go

```diff
@@ -603,6 +604,41 @@ var _ = Describe("[rfe_id:3503][crit:medium][vendor:cnv-qe@redhat.com][level:com
 				Expect(len(newVM.Spec.Template.Spec.Domain.Devices.Interfaces) == 1).To(Equal(true))
 			})
 		})
+		Context("when kubemacpool pod is re-started with 1 replica", func() {
+			BeforeEach(func() {
+				By("Changing replicas to 0 to start from the beginning")
+				err := changeManagerReplicas(0)
+				Expect(err).To(Succeed(), "should succeed changing replicas to 0")
+
+				By("Changing replicast to 1 to have just on pod with leader election")
+				err = changeManagerReplicas(1)
+				Expect(err).To(Succeed(), "should succeed changing replicas to 1")
+
+				By("Retrieving leader pod")
+				leaderPod, err := getLeaderPod()
+				Expect(err).To(Succeed(), "should succeed getting leader kubemacpool pod")
+
+				By("Seding SIGTERM signal to restart leader pod")
+				killCmd := []string{"exec", "-n", leaderPod.Namespace, leaderPod.Name, "--", "/bin/bash", "-c", "kill -s SIGTERM 1"}
+				output, err := cmd.Kubectl(killCmd...)
```

Comment:
I don't like it cause it'll be tempting for future me not to use it even when not necessary. But I found no other way, so I guess it's ok. :)

Reply:
What do you mean ?

Reply:
I mean it would be tempting to use "kubectl" cli commands instead of kubeclient instance..

Reply:
Wouldn't it be enough to delete the pod through kube-api go client?

Reply:
I you delete the pod you have new pod so leader election label is not there, what we want to test is that leader election is ok if the container inside the pod is restarted.